lib/src/acter_avatar.dart

```diff
@@ -56,9 +56,11 @@ class _ActerAvatar extends State<ActerAvatar> {
 
   // avoid re-run future when object state isn't changed.
   @override
-  void didChangeDependencies() {
-    super.didChangeDependencies();
-    setAvatar();
+  void didUpdateWidget(ActerAvatar oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.avatar != widget.avatar) {
```

Comment:
> Yeah, I was wondering how to fix that. Do you also have an idea how to solve the problem introduced by the delay of the future?

I think you missed adding ```initState()``` here which runs when widget is firstly initialized. So initially we won't have avatar because of async computation and fallback would be loaded. Then after that, this ```didUpdateWidget()``` serves as check if avatar object is changed, it'll automatically update the new state.